common/types/proposal.go

```diff
@@ -21,6 +24,25 @@ const (
 	ProposalIDSize = Hash32Length
 )
 
+// GenesisBlock returns the genesis block.
+func GenesisBlock() *Block {
```

Comment:
in my opinion, this gets us further from how it should be not closer. genesis block should be just an id. is there even any reason to store in database? in tortoise it is added to the state by re-generating it, which is also not good. i think it should be just a parameter in the genesis config.

Reply:
I intend to write a SMIP for the genesis flow in the next few days. I wasn't even aware that we have a `GenesisBlock`...

Does it have any functional purpose at the moment? What breaks if we remove it?

As far as I know we should only have a `GoldenATX` in epoch 0 which is used as the `PositioningATX` for the ATXs in epoch 1. That should be only an ID as well - eventually, it won't be a just a random number, but a hash of something we define in advance. I'll elaborate on that in the SMIP, but just want to know if I have to solve any problems if I want to remove the genesis **block**.

Reply:
tortoise depends on it so that it can start in verifying mode, otherwise, it will switch into full mode immediately. once this is implemented https://github.com/spacemeshos/go-spacemesh/issues/2985 that dependency will be irrelevant

Reply:
ok. now changed to use hardcoded hex strings for genesis ballot and block IDs.

> is there even any reason to store in database?

i tried not storing it in the database but that caused several tortoise test failures. i didn't want to spend time fixing them so leave them as is for now.